### PR TITLE
docs!: switch to Sphinx Book theme

### PR DESCRIPTION
**Pull request by redeboer**
While trying to embed some iPython widgets in the Jupyter notbooks with `nbsphinx` (it doesn't work well), I ran into [executablebooks.org](https://executablebooks.org/en/latest). Seems like a promising project. While using their [Jupyter Book](https://jupyterbook.org/intro.html) infrastructure may be a solution for rendering interactive Jupyter notebook cells, [sphinx-book-theme.rtfd.io](https://sphinx-book-theme.readthedocs.io) may be a nice middle ground for now. It seems to work well with [Thebe](https://sphinx-book-theme.readthedocs.io/en/latest/launch.html?highlight=notebook_interface#live-code-cells-with-thebe).

For sure the theme looks quite nice:
https://pwa--251.org.readthedocs.build/projects/expertsystem/en/251/

### CODE REVIEW

File: doc/contribute.rst [hunk 1]
```diff
@@ -322,8 +323,8 @@ Commit conventions
   summary based on the separate commits that constitute this PR. Leave the
   relevant commits in as bullet points. See the `commit history
   <https://github.com/ComPWA/expertsystem/commits/master>`_ for examples. This
-  comes in especially handy when `drafting a release <contribute:Milestones and
-  releases>`_!
```
Comment (redeboer) [inline doc/contribute.rst:326]: @spflueger This caused the error discussed earlier

File: doc/index.rst [hunk 2]
```diff
@@ -105,49 +107,45 @@ Workflow of the Expert System
...
 
-   2.2. *Clone* graphs while inserting concrete matching particles for the
-   intermediate edges (mainly adds the mass variable).
+  2.2. *Clone* graphs while inserting concrete matching particles for the
+  intermediate edges (mainly adds the mass variable).
 
-   2.3. *Validate* the complete graphs, so run all conservation law check that
-   were postponed from the first step.
+  2.3. *Validate* the complete graphs, so run all conservation law check that
+  were postponed from the first step.
```
Comment (spflueger) [inline doc/index.rst:128]: The grey bars on the left side are not my style. If that can be changed into a nested enumeration it would be nicer.

**Review by spflueger — APPROVED** [review]
Looks very neat! I just have one comment about an enumeration style

File: doc/index.rst (hunk 2 above)
Comment (redeboer) [reply doc/index.rst:128]: Ok I tried around a bit, but I can't get the bar to disappear, only with a nested **bullet** list. It seems to be a shortcoming of the Sphinx Book Theme

**Pull request merged**